packages/client/src/controllers/engine.ts

```diff
@@ -731,13 +686,16 @@ export class Engine extends IEngine {
 
   private onSessionRequest: EnginePrivate["onSessionRequest"] = async (topic, payload) => {
     // TODO(ilja) validation
-    const { methods } = await this.client.session.get(topic);
+    const { namespaces } = await this.client.session.get(topic);
     const { id, params } = payload;
     const { chainId, request } = params;
-    if (!methods.includes(params.request.method)) {
-      await this.sendError(id, topic, ERROR.UNAUTHORIZED_JSON_RPC_METHOD.format());
-    } else if (!params.chainId) {
+    const isChain = Boolean(chainId && namespaces.filter(n => n.chains.includes(chainId)).length);
```

Comment:
Could use `Array.some` here to get a boolean rather than going via implicit casting from length to boolean:

```suggestion
    const isChain = Boolean(chainId && namespaces.some(n => n.chains.includes(chainId)));
```